### PR TITLE
RM-9103 Integrate NUKE build script

### PR DESCRIPTION
**Pull request by a-ctor**
https://re-motion.atlassian.net/browse/RM-9103

### CODE REVIEW

File: Remotion/ObjectBinding/Web.IntegrationTests/App.config [hunk 1]
```diff
@@ -25,6 +23,7 @@
       screenshotDirectory=".\WebTestingOutput"
       logsDirectory=".\WebTestingOutput"
       closeBrowserWindowsOnSetUpAndTearDown="false"
+      cleanUpUnmatchedDownloadedFiles="false"
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.IntegrationTests/App.config:26]: Why the change?

File: Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/App.config [hunk 2]
```diff
@@ -26,6 +26,7 @@
       screenshotDirectory=".\WebTestingOutput"
       logsDirectory=".\WebTestingOutput"
       closeBrowserWindowsOnSetUpAndTearDown="false"
+      cleanUpUnmatchedDownloadedFiles="false"
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/App.config:29]: See comment in other web test app.config

File: Build/Shared.build.NetFramework.props [hunk 3]
```diff
@@ -3,7 +3,6 @@
   <!-- Settings for projects that continue to use classic csproj files (WebForms projects) -->
   <PropertyGroup>
     <LangVersion>12</LangVersion>
-    <DebugType>full</DebugType>
```
Comment (MichaelKetting) [inline Build/Shared.build.NetFramework.props:6]: please document a feature for the updated symbols information.

File: .nuke/parameters.ci-chrome.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "./build.schema.json",
+  "Solution": "Remotion.sln",
+  "Target": [ "CIBuild" ],
+  "TestConfigurations": [ "Debug", "Release" ],
+  "TestExecutionRuntimes": [ "EnforcedLocalMachine" ],
```
Comment (MichaelKetting) [inline .nuke/parameters.ci-chrome.json:6]: browser build uses docker. the old script had 
```
      <ExecutionRuntimes>Win_NET48+EnforcedLocalMachine</ExecutionRuntimes>
```

File: .nuke/parameters.ci-edge.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "./build.schema.json",
+  "Solution": "Remotion.sln",
+  "Target": [ "CIBuild" ],
+  "TestConfigurations": [ "Debug", "Release" ],
+  "TestExecutionRuntimes": [ "EnforcedLocalMachine" ],
```
Comment (MichaelKetting) [inline .nuke/parameters.ci-edge.json:6]: see ci_chrome

File: .nuke/parameters.ci-firefox.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "./build.schema.json",
+  "Solution": "Remotion.sln",
+  "Target": [ "CIBuild" ],
+  "TestConfigurations": [ "Debug", "Release" ],
+  "TestExecutionRuntimes": [ "EnforcedLocalMachine" ],
```
Comment (MichaelKetting) [inline .nuke/parameters.ci-firefox.json:6]: see ci_chrome

File: .nuke/parameters.ci.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  "$schema": "./build.schema.json",
+  "Solution": "Remotion.sln",
+  "Target": [ "CIBuild" ],
+  "TestConfigurations": [ "Debug" ],
+  "TestExecutionRuntimes": [ "Docker_Win_NET48", "EnforcedLocalMachine" ],
+  "TestPlatforms": [ "x86" ],
+  "TestTargetRuntimes": [ "NET48" ],
+  "TestBrowsers": [ "NoBrowser", "Chrome" ],
+  "TestSqlServers": [ "NoDB", "SqlServer2016" ],
+  "TestFilter": "TestCategory!=LongRunning"
```
Comment (MichaelKetting) [inline .nuke/parameters.ci.json:11]: do yoy have the nunit ticket to watch for me?

File: Build/Customizations/SqlServers.cs [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using Remotion.BuildScript.Test;
+
+namespace Customizations;
+
+public class SqlServers : TestDimension
```
Comment (MichaelKetting) [inline Build/Customizations/SqlServers.cs:21]: Rename "Databases"

**Review by MichaelKetting — APPROVED** [review]
some small details, don't need to see again

File: Remotion/ObjectBinding/Web.IntegrationTests/App.config (hunk 1 above)
Comment (a-ctor) [reply Remotion/ObjectBinding/Web.IntegrationTests/App.config:26]: There is no change in configuration its just an explicit default value to make patching the file easier.

File: Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/App.config (hunk 2 above)
Comment (a-ctor) [reply Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/App.config:29]: See other comment just an explicit default value to make the patching easier

File: .nuke/parameters.ci-chrome.json (hunk 4 above)
Comment (a-ctor) [reply .nuke/parameters.ci-chrome.json:6]: Okay maybe I misunderstood the hack with `EnforcedLocalMachine` but I thought that this setup runs the tests locally, but hosts the test site in the specified docker container. Otherwise `EnforceLocalMachine` is really bad naming.

The `WebTestSetup` class is used in the web tests to pass the required docker image name to the web tests. This is no longer hacked into test dimension matrix.

File: .nuke/parameters.ci.json (hunk 7 above)
Comment (a-ctor) [reply .nuke/parameters.ci.json:11]: https://github.com/nunit/nunit/issues/4589

File: Build/Customizations/SqlServers.cs (hunk 8 above)
Comment (a-ctor) [reply Build/Customizations/SqlServers.cs:21]: Well that is a breaking change and I would have needed that info before releasing a version ^^

File: Build/Build.cs [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+    projects.AddReleaseProject("Development.Mixins");
+    projects.AddReleaseProject("Development.Web");
+    projects.AddReleaseProject("Development.Sandboxing.NUnit2");
+    projects.AddReleaseProject("Development.UnitTesting.Compilation");
+    projects.AddReleaseProject("Development.UnitTesting.IsolatedCodeRunner");
+    projects.AddUnitTestProject("Development.UnitTests", normalTestConfiguration);
+    projects.AddUnitTestProject("Development.Moq.UnitTests", normalTestConfiguration);
+    projects.AddUnitTestProject("Development.Sandboxing.NUnit2.UnitTests", normalTestConfiguration);
+    projects.AddUnitTestProject("Development.UnitTesting.IsolatedCodeRunner.UnitTests", normalTestConfiguration);
+    projects.AddReleaseProject("Mixins.Core");
```
Comment (MichaelKetting) [inline Build/Build.cs:126]:
```suggestion
    projects.AddReleaseProject("Mixins.Core");
    projects.AddReleaseProject("Mixins.CrossReferencer");
```

File: Build/Build.cs [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+    projects.AddReleaseProject("Mixins.MixerTools");
+    projects.AddUnitTestProject("Mixins.UnitTests", normalTestConfiguration);
```
Comment (MichaelKetting) [inline Build/Build.cs:128]:
```suggestion
    projects.AddUnitTestProject("Mixins.UnitTests", normalTestConfiguration);
    projects.AddUnitTestProject("Mixins.CrossReferencer.UnitTests", normalTestConfiguration);
```
